deeplabcut/pose_estimation_pytorch/runners/snapshots.py

```diff
@@ -113,41 +113,43 @@ def update(self, epoch: int, state_dict: dict, last: bool = False) -> None:
         ):
             current_best = self.best()
             self._best_metric = metrics[self._key]
+            
+            # Save the new best model
             save_path = self.snapshot_path(epoch, best=True)
             parsed_state_dict = {
-                k: v
+                k: v 
                 for k, v in state_dict.items()
                 if self.save_optimizer_state or k != "optimizer"
             }
             torch.save(parsed_state_dict, save_path)
 
+            # Handle previous best model
             if current_best is not None:
-                # rename if the current best should have been saved, otherwise delete
                 if current_best.epochs % self.save_epochs == 0:
                     new_name = self.snapshot_path(epoch=current_best.epochs)
                     current_best.path.rename(new_name)
                 else:
                     current_best.path.unlink(missing_ok=False)
-            return
-
-        if not (last or epoch % self.save_epochs == 0):
-            return
-
+        else:
+            # Save regular snapshot if needed
+            should_save = last or epoch % self.save_epochs == 0
+            if should_save:
+                save_path = self.snapshot_path(epoch=epoch)
+                parsed_state_dict = {
+                    k: v
+                    for k, v in state_dict.items()
+                    if self.save_optimizer_state or k != "optimizer"
+                }
+                torch.save(parsed_state_dict, save_path)
```

Comment:
I think we can be more concise here:

```suggestion
        elif last or epoch % self.save_epochs == 0:
            # Save regular snapshot if needed
            save_path = self.snapshot_path(epoch=epoch)
            parsed_state_dict = {
                k: v
                for k, v in state_dict.items()
                if self.save_optimizer_state or k != "optimizer"
            }
            torch.save(parsed_state_dict, save_path)
```